src/pharmpy/plugins/nonmem/model.py

```diff
@@ -343,19 +343,20 @@ def _read_dataset(self, raw=False, parse_columns=tuple()):
             dtype=None if raw else self.datainfo.get_dtype_dict(),
         )
         # Let TIME be the idv in both $PK and $PRED models
-
         # Remove individuals without observations
-        try:
-            from pharmpy.modeling.data import get_observations
-
-            # This is a hack to be able to use the get_observations function
-            # before the dataset has been properly read in.
-            self._data_frame = df
-            have_obs = set(get_observations(self).index.unique(level=0))
-        except DatasetError:
-            pass
+        col_names = list(df.columns)
+        have_pk = self.internals.control_stream.get_pk_record()
+        if 'EVID' in col_names:
+            df_obs = df.astype({'EVID': 'float'}).query('EVID == 0')
+        elif 'MDV' in col_names:
+            df_obs = df.astype({'MDV': 'float'}).query('MDV == 0')
+        elif 'AMT' in col_names and have_pk:
+            df_obs = df.astype({'AMT': 'float'}).query('AMT == 0')
         else:
```

Comment:
Only raise here if have_pk. If not all rows are observations

Reply:
Thanks, I will change it.

Reply:
> From the NONMEM documentation it seems as if EVID is also only interpreted if $PK. MDV is unclear from the documentation, but seems to be general. Could be tested in a resmod model.

Yes, this is also my question, because I check the `parse_datainfo()` function https://github.com/pharmpy/pharmpy/blob/37144d60b0d988822c37b7f1462d6a071eb71200/src/pharmpy/plugins/nonmem/parsing.py#L610 I think our current way is checking the PK record for EVID, MDV and AMT. I think maybe we should apply same strategy here. I can have a test on the resmod model to see what happens.

Reply:
Great. If the MDV is not used in a $PRED model then the default should be to have all records as observations and only consider EVID, MDV and AMT if model has $PK

Reply:
Here is the result: ` (WARNING  13) WITH USER-WRITTEN PRED OR $PRED, NM-TRAN CANNOT APPEND THE
 MDV DATA ITEM.`
I think this warning indicates the MDV is not used in the $PRED model?